plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java

```diff
@@ -2562,4 +2570,89 @@ public static boolean isUniqueIndexForAttributes(
         }
         return false;
     }
+
+    public record ConnectivityParameters(
+        @Nullable String hostName,
+        @Nullable String hostPort,
+        @Nullable String databaseName,
+        @Nullable String userName,
+        @Nullable String server
+    ) {
+    }
+
+    @NotNull
+    private static ConnectivityParameters getExplicitConnectivityParameters(@NotNull DBPConnectionConfiguration configuration) {
+        return new ConnectivityParameters(
+            configuration.getHostName(),
+            configuration.getHostPort(),
+            CommonUtils.notEmptyOrDefault(configuration.getBootstrap().getDefaultCatalogName(), configuration.getDatabaseName()),
+            configuration.getUserName(),
+            configuration.getServerName()
+        );
+    }
+
+    @NotNull
+    public static ConnectivityParameters getConnectivityParameters(
+        @NotNull DBPConnectionConfiguration configuration,
+        @NotNull DBPDriver driver
+    ) throws DBException {
+        ConnectivityParameters explicitConfiguration = getExplicitConnectivityParameters(configuration);
+        return switch (configuration.getConfigurationType()) {
+            case MANUAL -> explicitConfiguration;
+            case URL -> {
+                String activeUrl = driver.getConnectionURL(configuration);
+                if (CommonUtils.isNotEmpty(activeUrl)) {
+                    ConnectivityParameters urlConnectivityParams = null;
+                    DBPConnectionConfiguration urlConfiguration = null;
+                    DatabaseURL.MetaURL metaURL = null;
+                    if (CommonUtils.isNotEmpty(driver.getSampleURL())) {
+                        urlConfiguration = DatabaseURL.extractConfigurationFromUrl(driver.getSampleURL(), activeUrl);
+                        if (urlConfiguration != null) {
+                            metaURL = DatabaseURL.parseSampleURL(driver.getSampleURL());
+                        }
+                    }
+                    if (urlConfiguration == null) {
+                        urlConfiguration = DatabaseURL.extractConfigurationFromUrl(DatabaseURL.GENERIC_URL_TEMPLATE, activeUrl);
+                        if (urlConfiguration != null) {
+                            metaURL = DatabaseURL.parseSampleURL(DatabaseURL.GENERIC_URL_TEMPLATE);
+                        }
+                    }
+                    if (urlConfiguration != null) {
+                        urlConnectivityParams = getExplicitConnectivityParameters(urlConfiguration);
+                    }
+                    if (urlConnectivityParams == null) {
```

Comment:
Add some description in method.
Which explain the logic and flow.

Also - it looks that this function is used in PostgreSQL only. Maybe move it there?